re-work settings

lordwelch opened the pull request:
My goal is to combine settings and cli parameters as much as possible and use JSON, TOML or another format that is more structured.

lordwelch committed:
Fix log dir

lordwelch committed:
Simplify --only-set-cv-key

lordwelch committed:
Add replacement settings

mizaki commented on comictaggerlib/main.py:
```diff
@@ -15,226 +15,144 @@
+    def initialize(self) -> argparse.Namespace:
+        opts, _ = self.initial_arg_parser.parse_known_args()
+        assert opts is not None
+        setup_logging(opts.verbose, opts.config.user_log_dir / "log")
```
The `/ "log"` is causing the /log/log issue.

lordwelch replied:
Done

mizaki commented on comictaggerlib/settings/cmdline.py:
```diff
@@ -0,0 +1,343 @@
+    # Ensure this stays up to date with register_options
+    parser.add_argument(
+        "--config",
+        help="Config directory defaults to ~/.ComicTagger\non Linux/Mac and %%APPDATA%% on Windows\n",
```
Linux is ~/.config/ComicTagger

lordwelch replied:
Done

lordwelch committed:
Update help for --config

lordwelch committed:
Update help

lordwelch committed:
Refactor cli.py into a class

lordwelch committed:
Fix wait_and_retry_on_rate_limit

lordwelch committed:
Cleanup

lordwelch committed:
Set logging level on comictalker

lordwelch committed:
Make logs use the .log extension

lordwelch committed:
Settings tests

lordwelch committed:
Generalize settings

Add comments and docstrings
Create parent directories when saving
Add merging to normalize_options
Change get_option to return if the value is the default value

lordwelch committed:
Add tests

lordwelch committed:
Update Settings

lordwelch committed:
Move test cases to the testing package

Add comments to tests

lordwelch committed:
Add comments to tests

lordwelch committed:
Fix --no-overwrite

lordwelch committed:
Fix setting cmdline arguments

lordwelch committed:
Split out settings functions

lordwelch committed:
Split settings out into a separate package

lordwelch committed:
Fail if an error occurs when loading settings

lordwelch committed:
Update to settngs 0.3.0

Use the namespace instead of a dictionary
Cleanup setting names
